### DISCUSSION TITLE
MakeFrom这个函数为什么不能传入已经绑定了其他FBO的纹理？“Note, the texture must not    * be bound to any frame buffer”

### DISCUSSION THREAD

**Discussion by hjm1fb** [Tencent/libpag#2193] [Q&A] [answered]
文件是pag.h

 /**
   * Creates a new PAGSurface from specified backend texture and origin. Note, the texture must not
   * be bound to any frame buffer. Passes true in the forAsyncThread parameter if the returned
   * PAGSurface will be used for asynchronous rendering. If passing true, the PAGSurface internally
   * creates a independent GPU context, and the caller can use semaphore objects to synchronise
   * content (see flushAndSignalSemaphore() and wait()), otherwise, it uses the GPU context on the
   * calling thread directly. Returns null if the texture is invalid.
   */
  static std::shared_ptr<PAGSurface> MakeFrom(const BackendTexture& texture, ImageOrigin origin,
                                              bool forAsyncThread = false);

我试了一下， 外面fbo上已经绑定了纹理A, 用纹理A以及FBO的宽高构造backTexture来做pag渲染，最后的确fbo上会有pag的图像。貌似没问题？

**Comment by domchen** [comment, marked as answer]
你是什么平台测试成功的？或者你检查一下是不是其实绑定别的FBO渲染完的时候你自己又解绑过？可能各个平台的GL校验规则不一样。正常来说如果一个纹理已经绑定到了另外一个FBO，那么再绑定第二个的时候，会报错无法绑定上。你传入的纹理我们需要在内部创建新的FBO绑定它来渲染。所以不能在外部已经绑定别的。为了最大化兼容也不建议这么操作。
